src/mlpack/core/metrics/bleu_score.hpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * @file core/metrics/bleu_score.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * BLEU, or the Bilingual Evaluation Understudy, is an algorithm for evaluating
+ * the quality of text which has been machine translated from one natural
+ * language to another. It can also be used to evaluate text generated for a
+ * suite of natural language processing tasks.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_METRICS_BLEU_SCORE_HPP
+#define MLPACK_CORE_METRICS_BLEU_SCORE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace metric {
+
+template <typename ElemType = float,
+          typename PrecisionType = std::vector<float>
+>
+class BLEU
+{
+ public:
+  /**
+   * Create an instance of BLEU class.
+   */
+  BLEU(const size_t maxOrder = 4);
```

Comment:
Can you add a comment for the `maxOrder` parameter?

Reply:
Done.